src/test/scala/io/iohk/ethereum/domain/TransactionSpec.scala

```diff
@@ -67,14 +88,26 @@ class TransactionSpec extends AnyFlatSpec with Matchers {
       .get
     val stx = SignedTransaction.apply(
       tx = tx,
-      signature = sig
+      // hacky change to make the test succeed without regressing the general workflow.
```

Comment:
at the time of coding, there wasn't , but the signer stuff will remove the need for the hack, since there should be a method to import the signature according to a given ruleset.

This behavior is also caused by the fact that, internally, Mantis uses the concept of ECDSASignature for two things:
- the actual crypto signature, which is unambiguous, and independant of ethereum/mantis
- the field r, s and v in the signed transaction (a signed transaction is the union of a Transaction and a ECDSASignature).

I'm wondering if we shouldn't introduce a EthererumEncodedSignature to properly differentiate between the two, since the v field value is converted between the crypto signature (possible value 0/1) and the rlp encoding (for exemple, between eip-155, it's 27/28, but after eip-155, it's 0/1 + chainId * 2 + 35, and for transaction type 1, it's back to 0/1)
Another possibility would be to only keep the unmodified ECDSASignature and doing the computation on the fly when rlp encoding/decoding. However it could possibly lead to the introduction of additional fields (encoding version, chainId, etc) , which would be dependant on the signature ruleset.